scipy/signal/spectral.py

```diff
@@ -826,8 +850,13 @@ def detrend_func(d):
             scale = 1.0 / win.sum()**2
         else:
             raise ValueError('Unknown scaling: %r' % scaling)
-    else:
-        scale = 1
+    elif mode == 'complex':
+        if scaling == 'density':
+            scale = np.sqrt(1.0 / (fs * (win*win).sum()))
+        elif scaling == 'spectrum':
+            scale = 1.0 / win.sum()
+        else:
+            raise ValueError('Unknown scaling: %r' % scaling)
```

Comment:
Is there a way to check that these scale factors are appropriate, e.g. by computing some short-length spectrogram in Octave or MATLAB and verifying `assert_allclose` with the expected result (which can be pasted into the tests)?